core/src/main/java/hudson/init/impl/InstallUncaughtExceptionHandler.java

```diff
@@ -37,5 +40,22 @@ public void reportException(Throwable e, ServletContext context, HttpServletRequ
                 }
             }
         });
+        Thread.setDefaultUncaughtExceptionHandler(new DefaultUncaughtExceptionHandler());
```

Comment:
May want to catch `SecurityException` in case Tomcat has decided to prevent you from setting this